server/main.py

```diff
@@ -153,6 +185,21 @@ async def room(websocket: WebSocket, room_name: str) -> None:
     """
     client = Client(websocket)
     await manager.connect(client, room_name)
+    initial_data = ConnectionData(**await client.receive())
+
+    if (initial_data.type == "connect") and (initial_data.data.room_code):
+        # If there is a room code, they have attempted to joined a room
+        if not manager.room_exists(initial_data.data.room_code):
```

Comment:
This condition will never be true because when calling `ConnectionManager.connect` the room will be created anyway. My suggestion would be to separate the code to create and join a room.